File: shareable-pedigree-link/python/src/shareable_pedigree_link/key_name.py
"""Pure helper to build the minted key's ``name`` field."""

from __future__ import annotations

_PEDIGREE_ID_PREFIX_LENGTH = 8


def build_key_name(pedigree_id: str, suffix: str) -> str:
    return f"share-{pedigree_id[:_PEDIGREE_ID_PREFIX_LENGTH]}-{suffix}"

File: shareable-pedigree-link/node/src/keyName.ts
const PEDIGREE_ID_PREFIX_LENGTH = 8;

export interface KeyNameRequest {
  readonly pedigreeId: string;
  readonly suffix: string;
}

export function buildKeyName(request: KeyNameRequest): string {
  const prefix = request.pedigreeId.slice(0, PEDIGREE_ID_PREFIX_LENGTH);
  return `share-${prefix}-${request.suffix}`;
}
